src/BenchmarkDotNet/Detectors/Cpu/Windows/PowershellWmiCpuDetector.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using BenchmarkDotNet.Helpers;
+using Perfolizer.Models;
+
+namespace BenchmarkDotNet.Detectors.Cpu.Windows;
+
+/// <summary>
+/// CPU information from output of the `wmic cpu get Name, NumberOfCores, NumberOfLogicalProcessors /Format:List` command.
+/// Windows only.
+/// </summary>
+internal class PowershellWmiCpuDetector : ICpuDetector
+{
+    private readonly string windowsPowershellPath =
+        $"{Environment.SystemDirectory}{Path.DirectorySeparatorChar}WindowsPowerShell{Path.DirectorySeparatorChar}" +
+        $"v1.0{Path.DirectorySeparatorChar}powershell.exe";
+
+    public bool IsApplicable() => OsDetector.IsWindows();
+
+    public CpuInfo? Detect()
+    {
+        if (!IsApplicable()) return null;
+
+        string programFiles = Environment.Is64BitOperatingSystem ? Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles) : Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
```

Comment:
The logic for determining the PowerShell location deserves its own method, or even a class, if we want to call it from another location.